index.js

```diff
@@ -624,9 +624,20 @@ function issecure(req, trustProxy) {
 
   // socket is localhost
   if (req.connection.remoteAddress === '127.0.0.1' ||
-      req.connection.remoteAddress === '::ffff:127.0.0.1'
+      req.connection.remoteAddress === '::ffff:127.0.0.1' ||
+      req.connection.remoteAddress === '::1'
   ) {
-    return true;
+    // if proxy is trusted; localhost connection is secure for sure
+    if (trustProxy === true) {
+      return true;
+    }
+
+    // proxy not explicitly trusted; no proxy means connection is secure
+    if (req.headers['x-forwarded-proto'] !== undefined) {
```

Comment:
Unfortunately this is also not a reliable test, because if the user does not trust the connection as a proxy, this header may not be set if not a proxy or may be forgable by the client.

Reply:
Sorry I am not sure what you mean, maybe I am missing some possible network configuration?

Since we detect a localhost connection we could trust the remote (reverse proxy or client) right?

Please correct me if I am wrong

Reply:
Not unless the user of this module specifies they trust it. This fixed a reported security vulnerability.

Reply:
Oh I see what you mean, so we can conclude that the only thing we can do is return true if the proxy is trusted and the connection is localhost, right?